### PR TITLE
Use process_p_u0_symbolic from MTK

### LINKED ISSUE #297: Symbolic remake does not work for nested systems
https://github.com/SciML/SciMLBase.jl/blob/d7e3d316a014967414f416efa35e4d1aca5458e5/src/remake.jl#L59

### PR DESCRIPTION
**Pull request by ValentinKaisermayer**
- Merge after #310 and https://github.com/SciML/ModelingToolkit.jl/pull/1941
- closes #297

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
This needs downstream test fixes to be setup.

**Comment by ValentinKaisermayer** [comment]
You mean a test of the symbolic remake in test/downstream?

**Comment by ChrisRackauckas** [comment]
This causes downstream test failures in DiffEqBase and ModelingToolkit. If those are intended, there should be a follow-up PR with test changes/fixes ready to be merged with the merge of this PR.

**Comment by ValentinKaisermayer** [comment]
1. The MTK test is marked as broken, which it no longer is with this PR. 

2. The DiffEqBase test is curios. The parameters are given as an empty dict, which is interpreted as a symbolic map.  Shouldn't it be `SciMLBase.NullParameters`?
```julia
prob1 = SplitODEProblem(f, f, u0, tspan, Dict(), callback = nothing)
prob2 = @inferred remake(prob1; u0 = prob1.u0 .+ 1)
```

**Comment by ChrisRackauckas** [comment]
Rebase this. What's left?

**Comment by ValentinKaisermayer** [comment]
Handling the empty varmap case. Sry, for being slow. I only have time for this on the weekends.

**Comment by ChrisRackauckas** [comment]
No worries, just checking on the health of this repo

**Comment by ValentinKaisermayer** [comment]
Needs follow-up PR in MTK in order to resolve:
```julia
symbolic remake with nested system: Error During Test at /home/runner/work/SciMLBase.jl/SciMLBase.jl/downstream/test/odesystem.jl:280
 Unexpected Pass
 Expression: prob_new.p == [4.0, 3.0, 1.0]
 Got correct result, please change to @test if no longer broken.

symbolic remake with nested system: Error During Test at /home/runner/work/SciMLBase.jl/SciMLBase.jl/downstream/test/odesystem.jl:281
 Unexpected Pass
 Expression: prob_new.u0 == [1.0, 0.0]
 Got correct result, please change to @test if no longer broken.
```

File: src/remake.jl [hunk 1]
```diff
@@ -133,8 +120,85 @@ function remake(prob::ODEProblem; f = missing,
...
+    end
+    if lcons === missing
+        lcons = prob.lcons
+    end
+    if ucons === missing
+        ucons = prob.ucons
+    end
+    if sense === missing
+        sense = prob.sense
+    end
```
Comment (ValentinKaisermayer) [inline src/remake.jl:183]: This is very basic for now and not even checks anything but it can be added later.

**Comment by ChrisRackauckas** [comment]
The last thing is just the empty dict handling now?

**Comment by ValentinKaisermayer** [comment]
> The last thing is just the empty dict handling now?

No that's in. Didn't the DiffEqBase test pass now?

**Comment by ChrisRackauckas** [comment]
Thanks! Missed that.

**Pull request merged**